File: LeetCode/java/1704_Determine_if_String_Halves_Are_Alike.java
class Solution {
    public boolean halvesAreAlike(String s) {
        int mid = s.length() / 2;
        int vowelsCnt1 = 0, vowelsCnt2 = 0;

        for (int i = 0; i < mid; i++) {
            if (s.charAt(i) == 'a' || s.charAt(i) == 'e' || s.charAt(i) == 'u' || s.charAt(i) == 'o' || s.charAt(i) == 'i' || s.charAt(i) == 'A' || s.charAt(i) == 'E' || s.charAt(i) == 'U' || s.charAt(i) == 'O' || s.charAt(i) == 'I') {
                vowelsCnt1++;
            }
        }

        for (int i = mid; i < s.length(); i++) {
            if (s.charAt(i) == 'a' || s.charAt(i) == 'e' || s.charAt(i) == 'u' || s.charAt(i) == 'o' || s.charAt(i) == 'i' || s.charAt(i) == 'A' || s.charAt(i) == 'E' || s.charAt(i) == 'U' || s.charAt(i) == 'O' || s.charAt(i) == 'I') {
                vowelsCnt2++;
            }
        }

        if (vowelsCnt1 == vowelsCnt2) {
            return true;
        }

        return false;
    }
}

File: LeetCode/cpp/1704_Determine_if_String_Halves_Are_Alike.cpp
class Solution {
public:
	bool halvesAreAlike(string s) {
		int mid = s.size() / 2;
		int vowels_cnt_1 = 0, vowels_cnt_2 = 0;

		for (int i = 0; i < mid; i++) {
			if (s[i] == 'a' || s[i] == 'e' || s[i] == 'u' || s[i] == 'o' || s[i] == 'i' || s[i] == 'A' || s[i] == 'E' || s[i] == 'U' || s[i] == 'O' || s[i] == 'I') {
				vowels_cnt_1++;
			}
		}

		for (int i = mid; i < s.size(); i++) {
			if (s[i] == 'a' || s[i] == 'e' || s[i] == 'u' || s[i] == 'o' || s[i] == 'i' || s[i] == 'A' || s[i] == 'E' || s[i] == 'U' || s[i] == 'O' || s[i] == 'I') {
				vowels_cnt_2++;
			}
		}

		if (vowels_cnt_1 == vowels_cnt_2) {
			return true;
		}

		return false;
	}
};